src/aws-cpp-sdk-core/include/smithy/interceptor/impl/ChecksumInterceptor.h

```diff
@@ -0,0 +1,209 @@
+                        }
+                    }
+                    else if (checksumAlgorithmName == "sha256")
+                    {
+                        if (modeledRequest.IsStreaming())
+                        {
+                            httpRequest->SetRequestHash(checksumAlgorithmName, Aws::MakeShared<Aws::Utils::Crypto::Sha256>(CHECKSUM_INTERCEPTOR_LOG_TAG));
+                        }
+                        else
+                        {
+                            const auto body = modeledRequest.GetBody()? modeledRequest.GetBody() : Aws::MakeShared<Aws::StringStream>(CHECKSUM_INTERCEPTOR_LOG_TAG, "");
+                            httpRequest->SetHeaderValue(checksumType, Aws::Utils::HashingUtils::Base64Encode(Aws::Utils::HashingUtils::CalculateSHA256(*body)));
+                        }
+                    }
+                    else if (checksumAlgorithmName == "sha1")
+                    {
+                        if (modeledRequest.IsStreaming())
+                        {
+                            httpRequest->SetRequestHash(checksumAlgorithmName, Aws::MakeShared<Aws::Utils::Crypto::Sha1>(CHECKSUM_INTERCEPTOR_LOG_TAG));
+                        }
+                        else
+                        {
+                            const auto body = modeledRequest.GetBody()? modeledRequest.GetBody() : Aws::MakeShared<Aws::StringStream>(CHECKSUM_INTERCEPTOR_LOG_TAG, "");
+                            httpRequest->SetHeaderValue(checksumType, Aws::Utils::HashingUtils::Base64Encode(Aws::Utils::HashingUtils::CalculateSHA1(*body)));
+                        }
+                    }
+                    else if (checksumAlgorithmName == "md5" && headers.find(Aws::Http::CONTENT_MD5_HEADER) == headers.end())
+                    {
+                        const auto body = modeledRequest.GetBody()? modeledRequest.GetBody() : Aws::MakeShared<Aws::StringStream>(CHECKSUM_INTERCEPTOR_LOG_TAG, "");
+                        httpRequest->SetHeaderValue(Aws::Http::CONTENT_MD5_HEADER, Aws::Utils::HashingUtils::Base64Encode(Aws::Utils::HashingUtils::CalculateMD5(*body)));
+                    }
+                    else if (headers.find(Aws::Http::CONTENT_MD5_HEADER) == headers.end())
+                    {
+                        AWS_LOGSTREAM_WARN(CHECKSUM_INTERCEPTOR_LOG_TAG, "Checksum algorithm: " << checksumAlgorithmName << " is not supported by SDK.");
+                    }
+                }
+
+                // Response checksums
+                if (modeledRequest.ShouldValidateResponseChecksum())
+                {
+                    for (const Aws::String& responseChecksumAlgorithmName : modeledRequest.GetResponseChecksumAlgorithmNames())
+                    {
+                        const auto lowered = Aws::Utils::StringUtils::ToLower(responseChecksumAlgorithmName.c_str());
```

Comment:
cleanup: can be condensed for readability/reusability.  One approach is we can use base class for different hash types and invoke a method on base type to add response validation hash from a const set of supported hash type clases

Reply:
from my other comment

> all of the code in this file is copied from [the existing implentation](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/client/AWSClient.cpp#L799). I will refactor the logic in the next PR. This PR more or less jsut refactors the code out to a single location before handling the actual logical refactor. So while we can address comments on the logic here, a lot will be changing to it in the next PR.

This code could be cleaned up in a lot of different ways, and we plan on it. Its actually the next PR im going to raise. I dont like how this is strucutred and we ill be moving away from this giant if-else block alltogether. This PR is more focused on moving it out of the client and compartmentalizing it, then cleaning it up